### ISSUE TITLE
rough projection of time needed to run M2C and CV for each root

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/preservation_catalog#1381]
TL;DR:  run M2C on the storage root with the most druid-versions, and CV on the storage root with the largest amount of GB per druid, record the timings.

Please read this if you do this ticket.

We *may* find ourselves crunched to get these 2 validations run on storage newly migrated to a storage brick (especially if it's the last root to be migrated to a particular storage brick - there will be data from the previously migrated roots already on the brick).  Note that the current plan is to migrate multiple roots on the first 6 weekends:

from Julian:
>Pending disk copy testing, the order for migration is:
>
> Week 1: services_disk02 (6220) and services_disk08 (8040)
> Week 2: services_disk03 (6220) and services_disk09(8040)
> Week 3: services_disk04 (6220) and services_disk10(8040)
> Week 4: services_disk05 (6220) and services_disk11(8040)
> Week 5: services_disk06 (6220) and services_disk12(8040)
> Week 6: services_disk07 (6220) and services_disk13(8040)
> Week 7: services_disk00 (6220) and services_disk14(8040) ß services_disk00 is *not* going into PresCat. It’s just moving off the 6220.
> [ First brick full. Second storage brick must be online now to continue migrations. ]
> Week 8: services_disk15 (8040)
> Week 9: services_disk16 (8040)
> Week 10: services_disk17 (8040)
> Week 11: services_disk18 (8040)
> Week 12: services_disk19 (8040)
> Week 13: services_disk20 (8040)
> Week 14: services_disk21 (8040)
> Week 15: services_disk22 (8040)
> Week 16: services_disk23 (8040)
> 
> Assuming we start on time at the start of March, we should be done by the end of June.

With this is mind, we'd like to get elapsed time timings for running M2C and CV for the data on each existing storage root as a rough maximum of what to expect in timings from the new bricks.  

**running M2C on the single root with "the most druid-versions" and CV on "the biggest size" per druid root might be a way to get these rough timings:**

The following data is from Julian, a couple months out of date.  Per https://github.com/sul-dlss/shared_configs/blob/preservation-catalog-prod/config/settings/production.yml, we are up to services-disk22 now (but it will NOT be the biggest or have the most druid-versions so we're ok for getting timings):

![image](https://user-images.githubusercontent.com/96775/75195499-e61c6280-570e-11ea-946a-e0b12be62b61.png)

![image](https://user-images.githubusercontent.com/96775/75195515-eddc0700-570e-11ea-8925-b068cc989abe.png)

You can run this by `cap prod ssh` and then running the M2C rake task here:  https://github.com/sul-dlss/preservation_catalog/#moab-to-catalog-m2c-existenceversion-check  and the CV rake task for the other root here:  https://github.com/sul-dlss/preservation_catalog/#checksum-validation-cv 

This info might ultimately live elsewhere, but it could go into a comment here for now.

**Comment by ndushay** [comment]
We have some ooooold data for this as well, but I believe it was single threaded, NOT with our current setup with multiple redis workers for each check:
- https://docs.google.com/spreadsheets/d/1Xvk02asCm75lj5eCOrruhT0i9T8r8kHpUYjoEvLKbqM/edit#gid=1710590652

- https://github.com/sul-dlss/preservation_catalog/wiki/Stats-Seeding

**Comment by ndushay** [comment]
The most druid-versions are on services-disk-02.   The biggest files per druid are on services-disk-17

**Comment by ndushay** [comment]
m2c on services-disk-02 was kicked off with rake task on preservation-catalog-prod-02

**Comment by ndushay** [comment]
cv on services-disk17 was kicked off with rake task from prescat-prod-03 (with nohup)

**Comment by ndushay** [comment]
John ran this query (from https://github.com/sul-dlss/preservation_catalog/tree/master/db#catalog-seeding-just-ran-for-the-first-time--how-long-did-it-take-to-crawl-each-storage-root-how-many-moabs-does-each-have-whats-the-average-moab-size, more or less)

```
> MoabStorageRoot.joins(:complete_moabs).group(:name).order('name asc').pluck(:name, '(max(complete_moabs.last_checksum_validation)-min(complete_moabs.last_checksum_validation)) total_cv_lag', '(max(complete_moabs.last_moab_validation)-min(complete_moabs.last_moab_validation)) total_mv_lag', 'count(complete_moabs.id) num_moabs', 'round(avg(complete_moabs.size)) avg_moab_size')
=> [["services-disk02", "93 days 02:01:33.700949", "758 days 14:13:51.431803", 132718, 0.251382425e9],
 ["services-disk03", "94 days 00:08:43.758312", "380 days 00:51:21.591129", 116884, 0.196529842e9],
 ["services-disk04", "94 days 00:08:44.544881", "465 days 15:33:17.577267", 129243, 0.212599043e9],
 ["services-disk05", "89 days 23:09:51.991243", "105 days 02:16:14.469979", 153456, 0.199985323e9],
 ["services-disk06", "90 days 08:33:12.957508", "91 days 09:21:47.867956", 88692, 0.366560016e9],
 ["services-disk07", "86 days 12:38:17.531989", "175 days 16:28:58.601742", 108799, 0.278889364e9],
 ["services-disk08", "87 days 15:32:14.171903", "466 days 07:37:43.345428", 64019, 0.397740265e9],
 ["services-disk09", "83 days 14:46:53.338501", "105 days 11:32:23.710979", 39532, 0.665315402e9],
 ["services-disk10", "94 days 01:15:06.810362", "222 days 14:01:54.107914", 53407, 0.499131054e9],
 ["services-disk11", "86 days 10:50:59.237835", "168 days 00:30:38.408788", 9122, 0.3133844352e10],
 ["services-disk12", "91 days 00:30:35.348529", "167 days 16:52:41.968025", 172858, 0.290646842e9],
 ["services-disk13", "90 days 11:41:29.073521", "161 days 15:22:17.000435", 83955, 0.662533694e9],
 ["services-disk14", "90 days 19:18:25.923652", "157 days 02:53:04.356297", 103842, 0.514721041e9],
 ["services-disk15", "90 days 10:57:10.550128", "167 days 02:20:29.399122", 62564, 0.91078292e9],
 ["services-disk16", "94 days 00:08:53.568164", "489 days 00:32:09.449065", 314162, 0.167368137e9],
 ["services-disk17", "20:01:33.336075", "20:01:33.332496", 15155, 0.2651064669e10],
 ["services-disk18", "94 days 09:21:48.375062", "192 days 18:18:02.880497", 30135, 0.1764801545e10],
 ["services-disk19", "94 days 04:13:19.678226", "223 days 16:45:01.230522", 53297, 0.921051945e9],
 ["services-disk20", "94 days 04:28:40.947305", "147 days 06:32:11.537612", 18048, 0.289992989e10],
 ["services-disk21", "68 days 21:51:02.464594", "63 days 23:58:37.874871", 10732, 0.4827827575e10],
 ["services-disk22", "12 days 17:07:21.597935", "12 days 17:07:21.597935", 5787, 0.4119009449e10]]
```

Which shows that CV for disk17, which is 37TB, took 20 hours.    Doing 2 disks per weekend, we would have 40-80TB for CV, so that becomes 40 hours ... which is too long for a weekend cutover.  We will need more workers / RAM / whatever for CV workers, or we can only do one root per weekend.

**Comment by ndushay** [comment]
closing this;  we can re-open if we need to do more for time projections
